File: 1000/CF2230B_Digit_String.py
def solve(s):
    n = len(s)

    prefix = [0] * n
    suffix = [0] * n

    c = 0
    mx = 0

    for i in range(n):
        if s[i] == '2':
            c += 1
        prefix[i] = c

    c = 0
    for i in range(n - 1, -1, -1):
        if s[i] == '1' or s[i] == '3':
            c += 1
        suffix[i] = c

    for i in range(n):
        mx = max(mx, prefix[i] + suffix[i])

    return n - mx


t = int(input())

for _ in range(t):
    s = input().strip()
    ans = solve(s)
    print(ans)

File: 1000/CF2230B_Digit_String.java

import java.util.Scanner;

public class CF2230B_Digit_String 
{
    public static int solve(String s)
    {
        int n=s.length();
        int []prefix=new int[n];
        int []suffix=new int[n];
        int c=0;
        int mx=0;
        for (int i=0;i<n;i++)
        {
            if (s.charAt(i)=='2')
            {
                c++;
            }
            prefix[i]=c;
        }
        c=0;
        for (int i=n-1;i>=0;i--)
        {
            if (s.charAt(i)=='1' || s.charAt(i)=='3')
            {
                c++;
            }
            suffix[i]=c;
        }
        for (int i=0;i<n;i++)
        {
            mx=Math.max(mx, prefix[i]+suffix[i]);
        }
        return n-mx;
    }
    public static void main(String[] args) 
    {
        Scanner sc=new Scanner(System.in);
        int n=sc.nextInt();
        for (int i=0;i<n;i++)
        {
            String s=sc.next();
            int ans=solve(s);
            System.out.println(ans);
        }
        sc.close();
    }    
}
